pkg/bootstrap/docker/openshift/ansible.go

```diff
@@ -0,0 +1,308 @@
+	return &ansibleRunner{
+		Helper:       h,
+		KubeClient:   kubeClient,
+		ImageStreams: imageStreams,
+		Prefix:       prefix,
+		Namespace:    namespace,
+	}
+}
+func newAnsibleInventoryParams() ansibleInventoryParams {
+	return ansibleInventoryParams{
+		ansibleLoggingInventoryParams: ansibleLoggingInventoryParams{},
+		ansibleMetricsInventoryParams: ansibleMetricsInventoryParams{},
+	}
+}
+
+// writeInventoryToHost generates the inventory file given the parameters and writes
+// the inventory file to a temp directory on the host
+// return the basename of the inventory file
+func (r *ansibleRunner) uploadInventoryToHost(inventoryTemplate string, params ansibleInventoryParams) (string, error) {
+	inventory, err := generateAnsibleInventory(inventoryTemplate, params, r.ImageStreams)
+	if err != nil {
+		return "", err
+	}
+	file, err := ioutil.TempFile("", "openshift-inventory")
+	if err != nil {
+		return "", err
+	}
+	_, err = file.WriteString(inventory)
+	if err != nil {
+		return "", err
+	}
+	err = file.Close()
+	if err != nil {
+		return "", err
+	}
+	glog.V(1).Infof("Wrote inventory to local file: %s", file.Name())
+	dest := fmt.Sprintf("%s/%s.inventory", host.DefaultConfigDir, r.Prefix)
+	glog.V(1).Infof("Uploading file %s to host destination: %s", file.Name(), dest)
+	err = r.Helper.hostHelper.UploadFileToContainer(file.Name(), dest)
+	if err != nil {
+		return "", err
+	}
+	return path.Base(dest), nil
+}
+
+// generateAnsibleInventory and return the content as a string
+func generateAnsibleInventory(inventoryTemplate string, params ansibleInventoryParams, imageStreams string) (string, error) {
+
+	// set the deploymentType
+	if imageStreams == imageStreamCentos {
+		params.OSEDeploymentType = deploymentTypeOrigin
+	} else {
+		params.OSEDeploymentType = deploymentTypeOCP
+	}
+	t, err := template.New("").Parse(inventoryTemplate)
+	if err != nil {
+		return "", errors.NewError("Unable to parse ansible inventory template").WithCause(err)
+	}
+
+	inventory := &bytes.Buffer{}
+	err = t.Execute(inventory, params)
+	if err != nil {
+		return "", errors.NewError("Unable to substitute ansible params into the inventory template: %s", params).WithCause(err)
+	}
+	if glog.V(1) {
+		glog.V(1).Infof("Generated ansible inventory:\n %s\n", inventory.String())
+	}
+	return inventory.String(), nil
+
+}
+
+func (r *ansibleRunner) createServiceAccount(namespace string) error {
+	serviceAccount := &kapi.ServiceAccount{}
+	serviceAccount.Name = openshiftAnsibleServiceAccount
+	_, err := r.KubeClient.Core().ServiceAccounts(namespace).Create(serviceAccount)
+	if err != nil && !kapierrors.IsAlreadyExists(err) {
+		return errors.NewError(fmt.Sprintf("cannot create %s service account", serviceAccount.Name)).WithCause(err).WithDetails(r.Helper.OriginLog())
+	}
+	// Add privileged SCC to serviceAccount
+	if err = AddSCCToServiceAccount(r.KubeClient, "privileged", serviceAccount.Name, namespace); err != nil {
```

Comment:
We currently don't delete SA's used to do setup work. I don't see much benefit in doing so, as long as they're in namespaces that are not usually visible to regular users.